### PR TITLE
Added Signup, Login and Home pages; Basic sctructure

### PR DESCRIPTION
**Pull request by vikatitova**

### CODE REVIEW

File: src/components/AuthForm.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    const [validated, setValidated] = useState(false);
+    const [formInput, setFormInput] = useState({
+        email: '',
+        password: '',
+    });
+
+    const handleSubmit = (event: any) => {
+        const form = event.currentTarget;
+        if (form.checkValidity() === false) {
+            event.preventDefault();
```
Comment (ZubelRuslan) [inline src/components/AuthForm.tsx]: You can add common `event.preventDefault()` for two your scenarios (only move this before `if(){}`)

File: src/hooks/auth.hook.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        setUserId(null);
+        localStorage.removeItem(storageName);
+    }, []);
+
+    useEffect(() => {
+        // const data = JSON.parse(localStorage.getItem(storageName));
+        // if (data && data.token) {
+        //     login(data.token, data.userId);
+        // }
+        // setReady(true);
```
Comment (ZubelRuslan) [inline src/hooks/auth.hook.tsx:34]: remove please

File: src/react-app-env.d.ts [hunk 3]
```diff
@@ -1 +1,2 @@
 /// <reference types="react-scripts" />
+// declare module 'react-router-dom';
```
Comment (ZubelRuslan) [inline src/react-app-env.d.ts]: ?

File: src/routes.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+            </Route>
+            <Route path='/signup' exact>
+                <AuthForm />
+            </Route>
+            <Route path='/login' exact>
+                <AuthForm />
+            </Route>
+            <Redirect to='/' />
+        </Switch>
+    );
```
Comment (ZubelRuslan) [inline src/routes.tsx]: Please implement private route wrapper. Because now you have duplicated routes

File: src/utils.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    localStorage.setItem(TOKEN_KEY, 'TestLogin');
+};
+
+export const logout = () => {
+    localStorage.removeItem(TOKEN_KEY);
+};
+
+export const isLogin = () => {
+    return true;
+    // if (localStorage.getItem(TOKEN_KEY)) {
```
Comment (ZubelRuslan) [inline src/utils.ts]: Uncomment and refactor. `() => !!localStorage.getItem(TOKEN_KEY)`

**Review by ZubelRuslan — CHANGES_REQUESTED** [review]
I have asked you to use minimum one class component. Please refactor one of your components

File: src/components/AuthForm.tsx (hunk 1 above)
Comment (vikatitova) [reply src/components/AuthForm.tsx]: Added `event.preventDefault();` before if statement

File: src/hooks/auth.hook.tsx (hunk 2 above)
Comment (vikatitova) [reply src/hooks/auth.hook.tsx:34]: I will refactor this code with next pr's

File: src/react-app-env.d.ts (hunk 3 above)
Comment (vikatitova) [reply src/react-app-env.d.ts]: Yes this line can be removed becasuse I installed "@types/..." dependencies

File: src/utils.ts (hunk 5 above)
Comment (vikatitova) [reply src/utils.ts]: Done :)

File: src/hooks/auth.hook.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        setUserId(id);
+
+        localStorage.setItem(
+            storageName,
+            JSON.stringify({
+                userId: id,
+                token: jwtToken,
+            })
+        );
+    }, []);
```
Comment (ZubelRuslan) [inline src/hooks/auth.hook.tsx:21]: Better to store this data in separate localStorage fields. In your application you should parse JSON everytime now. But this is a small mistake

**Review by ZubelRuslan — APPROVED** [review]
Approved

**Pull request merged**